### PR TITLE
Bt 2973 | upload to playstore

### PR DESCRIPTION
**Pull request by GargShubhika**

### CODE REVIEW

**Review by ajaychoudhary-hotstar — APPROVED** [review]

File: .github/workflows/hotstar-android-build-workflow.yml [hunk 1]
```diff
@@ -74,7 +82,14 @@ jobs:
         run: . fetch_config secret/non-prod/clients/cloud && . ./gocd-setup.sh && cd scripts && chmod +x ./default_config_generator.sh && ./default_config_generator.sh
 
       - name: Generate Build
-        run: . fetch_config secret/non-prod/clients/cloud && cd scripts && ./fastlane-run.sh build_app_debug && cd ../ && echo "APP=$(ls -d app/build/outputs/apk/${BUILD_ENV}${BUILD_BUSINESS}/$BUILD_TYPE/hotstar*)" >> $GITHUB_ENV
+        run: . fetch_config secret/non-prod/clients/cloud && cd scripts && ./fastlane-run.sh build_app_debug && cd ../ && echo "APP=$(ls -d app/build/outputs/apk/${BUILD_ENV}${BUILD_BUSINESS}/$BUILD_TYPE/hotstar*)" >> $GITHUB_ENV && echo "SUPPLY_AAB=$(ls -d app/build/outputs/bundle/${BUILD_ENV}${BUILD_BUSINESS}Release/hotstar*)" >> $GITHUB_ENV
+
+      - name: Upload to Play Store
+        if: ${{ inputs.BUILD_TYPE == 'release' && inputs.BUILD_ENV == 'prod' && inputs.BUILD_BUSINESS == 'Mea' && startsWith(github.ref, 'refs/heads/release-') && inputs.UPLOAD_TO_PLAYSTORE == 'true' }}
```
Comment (ajaychoudhary-hotstar) [inline .github/workflows/hotstar-android-build-workflow.yml:88]: use only && inputs.UPLOAD_TO_PLAYSTORE no need to use == as this is already boolean

File: .github/workflows/hotstar-android-multibuild-workflow.yml [hunk 2]
```diff
@@ -35,5 +40,9 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Run Multi Builds
-        run: . fetch_config secret/non-prod/clients/multi-country-builds && cd scripts && ./run.sh
-  
\ No newline at end of file
+        run: . fetch_config secret/non-prod/clients/multi-country-builds && cd scripts && ./run.sh && echo "SUPPLY_AAB=$(ls -d app/build/outputs/bundle/${BUILD_ENV}${BUILD_BUSINESS}Release/hotstar*)" >> $GITHUB_ENV
```
Comment (ajaychoudhary-hotstar) [inline .github/workflows/hotstar-android-multibuild-workflow.yml:43]: no, this change is not required here, just add this parameter. Other steps/changes are not required

File: .github/workflows/hotstar-android-build-workflow.yml [hunk 3]
```diff
@@ -117,7 +132,8 @@ jobs:
         working-directory: pe-infra-scripts/perf-lab
         if: ${{ github.event.repository.name == 'hotstar-android-livingroom' }}
         run: python3 mozarkTest.py mozarkEnvVariablesFireTVX.json MEA Upload $(ls -d hotstarX*)
-
+
```
Comment (ajaychoudhary-hotstar) [inline .github/workflows/hotstar-android-build-workflow.yml:135]: undo this change

File: .github/workflows/hotstar-android-build-workflow.yml [hunk 4]
```diff
@@ -85,7 +100,7 @@ jobs:
       - name: Getting version details
         id: version
         run: echo "::set-output name=BUILD_VERSION::$(echo $APP | egrep -o '[0-9]+.[0-9]+.[0-9]+.[0-9]+-[0-9]+')"
-  
+
```
Comment (ajaychoudhary-hotstar) [inline .github/workflows/hotstar-android-build-workflow.yml:103]: same here, undo this

File: .github/workflows/hotstar-android-build-workflow.yml [hunk 5]
```diff
@@ -74,7 +82,14 @@ jobs:
...
 
       - name: Generate Build
-        run: . fetch_config secret/non-prod/clients/cloud && cd scripts && ./fastlane-run.sh build_app_debug && cd ../ && echo "APP=$(ls -d app/build/outputs/apk/${BUILD_ENV}${BUILD_BUSINESS}/$BUILD_TYPE/hotstar*)" >> $GITHUB_ENV
+        run: . fetch_config secret/non-prod/clients/cloud && cd scripts && ./fastlane-run.sh build_app_debug && cd ../ && echo "APP=$(ls -d app/build/outputs/apk/${BUILD_ENV}${BUILD_BUSINESS}/$BUILD_TYPE/hotstar*)" >> $GITHUB_ENV && echo "SUPPLY_AAB=$(ls -d app/build/outputs/bundle/${BUILD_ENV}${BUILD_BUSINESS}Release/hotstar*)" >> $GITHUB_ENV
+
+      - name: Upload to Play Store
+        if: ${{ inputs.BUILD_TYPE == 'release' && inputs.BUILD_ENV == 'prod' && inputs.BUILD_BUSINESS == 'Mea' && startsWith(github.ref, 'refs/heads/release-') && inputs.UPLOAD_TO_PLAYSTORE == 'true' }}
+        run: . fetch_config secret/non-prod/clients/cloud && . fetch_config secret/non-prod/android/${{ github.event.repository.name }}/playstore/${BUILD_BUSINESS}/package  && cd scripts  && ./fastlane-run.sh upload_to_playstore
```
Comment (ajaychoudhary-hotstar) [inline .github/workflows/hotstar-android-build-workflow.yml:89]: we can use . fetch_config secret/non-prod/clients/cloud,secret/non-prod/android/${{ github.event.repository.name }}/playstore/${BUILD_BUSINESS}/package

Comment (ajaychoudhary-hotstar) [reply .github/workflows/hotstar-android-build-workflow.yml:89]: means comma separated will work here instead of multiple fetch config

Comment (GargShubhika) [reply .github/workflows/hotstar-android-build-workflow.yml:89]: resolved

File: .github/workflows/hotstar-android-build-workflow.yml [hunk 6]
```diff
@@ -74,7 +80,13 @@ jobs:
         run: . fetch_config secret/non-prod/clients/cloud && . ./gocd-setup.sh && cd scripts && chmod +x ./default_config_generator.sh && ./default_config_generator.sh
 
       - name: Generate Build
-        run: . fetch_config secret/non-prod/clients/cloud && cd scripts && ./fastlane-run.sh build_app_debug && cd ../ && echo "APP=$(ls -d app/build/outputs/apk/${BUILD_ENV}${BUILD_BUSINESS}/$BUILD_TYPE/hotstar*)" >> $GITHUB_ENV
+        run: . fetch_config secret/non-prod/clients/cloud && cd scripts && ./fastlane-run.sh build_app_debug && cd ../ && echo "APP=$(ls -d app/build/outputs/apk/${BUILD_ENV}${BUILD_BUSINESS}/$BUILD_TYPE/hotstar*)" >> $GITHUB_ENV && echo "SUPPLY_AAB=$(ls -d app/build/outputs/bundle/${BUILD_ENV}${BUILD_BUSINESS}Release/hotstar*)" >> $GITHUB_ENV
+
+      - name: Upload to Play Store
+        if: ${{ inputs.BUILD_TYPE == 'release' && inputs.BUILD_ENV == 'prod' && inputs.BUILD_BUSINESS == 'Mea' && startsWith(github.ref, 'refs/heads/release-') && inputs.UPLOAD_TO_PLAYSTORE  }}
```
Comment (ajaychoudhary-hotstar) [inline .github/workflows/hotstar-android-build-workflow.yml:86]: I think there is extra space here. I should be single space. This is a minor thing but sometime GitHub actions don't take it well.

**Review by ajaychoudhary-hotstar — APPROVED** [review]
